### PR TITLE
Port kernel features support to fedora

### PR DESCRIPTION
**Pull request by rvykydal**
This is another take on https://github.com/rhinstaller/anaconda/pull/4858.
The original PR was updated by calling the match_available_packages in UI from the dnf manager / payload module (https://github.com/rhinstaller/anaconda/pull/4858#issuecomment-1653092036).

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
Can we also get a setting for forcing a particular variant in the profile configuration for particular architectures?

**Comment by jkonecny12** [comment]
I think you can extend it with another PR. However, we have to focus only on the high priority required parts.

File: pyanaconda/ui/gui/spokes/software_selection.py [hunk 1]
```diff
@@ -291,13 +314,52 @@ def _clear_listbox(self, listbox):
             listbox.remove(child)
             del child
 
+    def _refresh_kernel_features(self):
+        """Display options for selecting kernel features."""
+
+        # Only showing parts of kernel box relevant for current system.
+        self._available_kernels = get_available_kernel_features(self.payload.proxy)
+
+        if not conf.ui.show_kernel_options:
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/software_selection.py:323]: I would change this to
```
show_kernels = False
if conf.ui.show_kernels_options:
```

**Review by jkonecny12 — APPROVED** [review]
Other than the nitpick, looks great to me. Thanks!

File: pyanaconda/ui/gui/spokes/software_selection.py (hunk 1 above)
Comment (rvykydal) [reply pyanaconda/ui/gui/spokes/software_selection.py]: Updated

**Review by jkonecny12 — APPROVED** [review]

**Review by M4rtinK — APPROVED** [review]
Looks good to me as well.

**Pull request merged**